### PR TITLE
Refactor Controller:OutdoorAir functions

### PR DESCRIPTION
**Pull request by mjwitte**
## Pull request overview

Partially addresses #4633 and more.  See [design doc](https://github.com/NREL/EnergyPlus/blob/develop/design/FY2016/OAControllerMakeover-Issue4663.md) for details.

This pull request is the initial refactoring of `MixedAir::CalcOAController` and related functions.  
- Portions of `CalcOAController` have been split out into `CalcOAEconomizer` and `CalcMechVentController`.  
- Merged `DataAirloop::OAControllerInfo` into `MixedAir::OAController`.
- Deleted stray using statements.
- `CalcOAController`, `CalcOAEconomizer`, `SizeOAController`, `UpdateOAController` and `Checksetpoints` are now member functions of `OAControllerProps`.
- `ReportOAController` was an empty function and was deleted.
- `SetOAControllerData` was used by `HVACStandAloneERV` to populate `OAController objects` for `ZoneHVAC:EnergyRecoveryVentilator:Controller`. This function was deleted. `HVACStandAloneERV` now fills the `OAController` data directly.
- Converted some parts of `InitOAController` to use a single reference instead of full variables like `OAController( ControllerNum )`.
- In `CalcMechVentController` eliminated loops over all People objects just to count up occupants in one zone (both nominal and current total occupants are available from other data.).  Still needed to keep [one of these loops](https://github.com/NREL/EnergyPlus/blob/0fce018d6514b182caa183b16a2280d6267fb063/src/EnergyPlus/MixedAir.cc#L3881-L3886) for `SystemOAMethod == SOAM_ProportionalControlDesOcc` that adds up the CO2 output for the design number of people at the current activity level.  That could probably be collected elsewhere also to avoid extra steps here.

This PR should produce no diffs.  Then a new branch and PR will make the needed functional changes.
### Work Checklist

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
- [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
- [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
  - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
  - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
  - NewFeature: This pull request includes code to add a new feature to EnergyPlus
  - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
  - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog
### Review Checklist

This will not be exhaustively relevant to every PR.
- [x] Code style (parentheses padding, variable names)
- [x] Functional code review (it has to work!)
- [ ] If defect, results of running current develop vs this branch should exhibit the fix
- [x] CI status: all green or justified
- [ ] Performance: CI Linux results include performance check -- verify this
- [ ] Unit Test(s)
- C++ checks:
  - [ ] Argument types
  - [ ] If any virtual classes, ensure virtual destructor included, other things
- IDD changes:
  - [ ] Verify naming conventions and styles, memos and notes and defaults
  - [ ] If transition, add rules to spreadsheet
  - [ ] If transition, add transition source
  - [ ] If transition, update idfs
- [ ] If new idf included, locally check the err file and other outputs
- [ ] Documentation changes in place
- [ ] ExpandObjects changes??
- [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

File: design/FY2016/OAControllerMakeover-Issue4663.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+**M.J. Witte, GARD Analytics, Inc.**
+
+ - April 22, 2016
+
+
+## Background ##
+
+The current combination of Controller:OutdoorAir plus Controller:MechanicalVentilation an be confusing. Common frustrations that users encounter with these objects include:
+
```
Comment (rraustad) [inline design/FY2016/OAControllerMakeover-Issue4663.md]: ... an be confusing.

File: design/FY2016/OAControllerMakeover-Issue4663.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+```
+    ALWAYS_ON,               !- Availability Schedule Name
+    Yes,                     !- Demand Controlled Ventilation
+    VentilationRateProcedure,!- System Outdoor Air Method
+```
+
+2. Rename `Controller:MechanicalVentilation` to `Controller:OutdoorAir:ZoneList` and make it optional.  If all of the zones have a Sizing:Zone object, then the designspec info will be taken from there.
+
+3. Change the autosizing for `Controller:OutdoorAir` Minimum Outdoor Air Flow Rate to be zero if one of the advanced methods (VRP, DCV, IAQ) of control is selected.  *Or maybe autosize to the non-per-person flow?*
+
```
Comment (rraustad) [inline design/FY2016/OAControllerMakeover-Issue4663.md:125]: OK, so here minimum OA flow will be set by VRP, DCV, or IAQ.

File: design/FY2016/OAControllerMakeover-Issue4663.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+4. Make the Maximum Fraction of Outdoor Air Schedule Name be king - OA fraction can never be greater than the current schedule value.
+
```
Comment (rraustad) [inline design/FY2016/OAControllerMakeover-Issue4663.md:127]: OK with this.

File: design/FY2016/OAControllerMakeover-Issue4663.md [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+5. Change the advanced methods (VRP, DCV, IAQ) to use the `DesignSpecification:OutdoorAir` Outdoor Air Flow Rate Fraction Schedule.  Currently it is ignored.
+
```
Comment (rraustad) [inline design/FY2016/OAControllerMakeover-Issue4663.md:129]: Unsure here. How can a user specify the OA fraction when they request it be calculated by the program?

File: design/FY2016/OAControllerMakeover-Issue4663.md [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+6. Add a new outdoor air control type = "100PercentOA" which always delivers 100% OA.
```
Comment (rraustad) [inline design/FY2016/OAControllerMakeover-Issue4663.md]: So this can't be done now?

File: design/FY2016/OAControllerMakeover-Issue4663.md [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+
+7. Change "Time of Day Economizer Control Schedule" to apply to any type of economizer control, and add a new Economizer Control Type = TimeOfDay.
```
Comment (rraustad) [inline design/FY2016/OAControllerMakeover-Issue4663.md]: I like this one. This schedule could be useful for other reasons.

File: design/FY2016/OAControllerMakeover-Issue4663.md [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+
+8. Reorder and rename some of the fields in Controller:OutdoorAir.
```
Comment (rraustad) [inline design/FY2016/OAControllerMakeover-Issue4663.md]: Important for maintenance.

File: src/EnergyPlus/BranchInputManager.cc [hunk 8]
```diff
@@ -607,65 +606,6 @@ namespace BranchInputManager {
 		return GetAirBranchIndex;
 	}
```
Comment (rraustad) [inline src/EnergyPlus/BranchInputManager.cc:608]: The remaining changes will help to simplify code where air flow is dictated by the system and not the ductwork. Move forward with this.

File: design/FY2016/OAControllerMakeover-Issue4663.md (hunk 4 above)
Comment (mjwitte) [inline design/FY2016/OAControllerMakeover-Issue4663.md:129]: It's a multiplier on the base OA flow rate, so if it's zero then OA should go to zero.  ZoneSum will have no problem with this, but I will need to test the other methods to make sure the logic works for them.  It does raise a question for IAQ method.

File: design/FY2016/OAControllerMakeover-Issue4663.md (hunk 5 above)
Comment (mjwitte) [inline design/FY2016/OAControllerMakeover-Issue4663.md]: You can, but it's not obvious.  You have to set Min OA and Max OA equal to the max system flow rate, and if it's a VAV system, set to Proportional minimum.  Or, you could set the Min OA Fraction schedule to 1.0.  What most users try first is setting the Min OA Flow Rate Schedule to 1.0 and autosizing and that doesn't get you to 100% usually.

**Comment by Myoldmopar** [comment]
@mjwitte with #5582 merged in, I pulled develop into this branch, and updated the regression baseline branch to look back at develop.  CI should pick this up later today and still be clean I'd assume since it was all green before.

**Comment by mjwitte** [comment]
@Myoldmopar Thanks.  Since I haven't pushed up any actual code changes for this yet (just an outdated design doc at the moment), it better come back clean!

**Comment by mjwitte** [comment]
@Myoldmopar The goal for this PR was no difs.  The last commit moved a couple of calculations so that all things economizer would be in CalcOAEconomizer.  This is producing tiny diffs on the order of 1e-11 in two test files.

**Comment by mjwitte** [comment]
@Myoldmopar Ran a full annual test suite and found big diffs in AirEconomizerWithMaxMinOAFractions, HospitalBaseline, HospitalBaselineReheatReportEMS, and RefBldgHospitalNew2004_Chicago.  Reviewing those now.  Most appear to be tiny, and there are no tablediffs, but I still want to understand what changed.

**Comment by mjwitte** [comment]
@Myoldmopar @wfbuhl @rraustad Just pushed up the last of the refactoring changes for OAController functions.  This should come back with no diffs.  I've run local annual simulations for about 15 files with StandaloneERV and others with VRP or DCV in the name.  All came back with no diffs.  The changes in MixedAir.cc are too many to view on (wimpy) github, so you'll have to diff it locally.  Or just review it as-is and see if you like the approach.  Once this is reviewed and merged, then I will start a new branch to fix the various control issues; there the number of lines changed should be much smaller.

**Comment by wfbuhl** [comment]
Looks good to me

**Comment by Myoldmopar** [comment]
I looked at this locally.  I'm pumped about the refactoring.  Once you ignore all the simple search/replace style changes, the refactoring looks great.  1 CI failure is unrelated to this work.  @wfbuhl likes the changes as well.  A 0.02% increase in valgrind results are going to go in without a question on my end.  I'm going to merge.

**Pull request merged**